### DISCUSSION TITLE
The Go Micro Community

### DISCUSSION THREAD

**Discussion by asim** [micro/go-micro#2473] [General]
I have now created the org https://github.com/go-micro. This is likely where we'll move community led work and extract things like plugins and examples once more. 

Edit: Announcement post related to this https://medium.com/@asimaslam/the-go-micro-community-83cc65b1796e

**Comment by xpunch** [comment]
Please invite me into this org, so I can move some personal project into this org, like [go micro example](https://github.com/xpunch/go-micro-example), [go micro dashboard](https://github.com/xpunch/go-micro-dashboard).

**Comment by asim** [comment]
I will invite you soon. Already starting a move of things.

**Comment by asim** [comment]
Invited you as an owner of the org. I've put cli/dashboard/plugins/examples there. You can do whatever you want with it... I want to now remove these from the go-micro project itself

**Comment by asim** [comment]
Before I delete plugins, can you just make sure its all good in github.com/go-micro/plugins so people can easily switch over

**Reply by xpunch** [reply]
Before delete plugins, we need to tag all plugins using [release.sh](https://github.com/asim/go-micro/blob/master/plugins/release.sh). So users can still get plugin modules through given version.

**Reply by xpunch** [reply]
I'm handling the github.com/go-micro/plugins import issues.

**Reply by xpunch** [reply]
github.com/go-micro/plugins works fine now, but plugins and examples different between go micro versions. I think we should separate plugins and examples into different version folders, like what I've done in [go micro example](https://github.com/xpunch/go-micro-example), [go micro plugins](https://github.com/xpunch/go-micro-plugins).

**Reply by asim** [reply]
I'll let you handle the details of that. Anything in the go-micro org you can make changes without my approval

**Reply by xpunch** [reply]
OK

**Reply by xpunch** [reply]
github.com/go-micro/plugins seams work fine now, should we tag latest plugins in asim/go-micro and then delete it?

**Reply by asim** [reply]
Yea the time has come. Tag and delete.

**Comment by hhstore** [comment]
- What is the latest status of the `go-micro v4` version now? Can it be used in `production`? 
- Looks like `go-micro` is being maintained again? This is good.
- 😄

**Reply by xpunch** [reply]
Go micro v4 is stable, it's the successor of go micro v3 and includes latest bug fixings, so it can be used in production. The main change of go micro v4 is it used new vanity url, it could a breaking changes when migrate from v2/v3.  Go micro v3 is also stable, but only accept bug fixings.

**Reply by hhstore** [reply]
> Go micro v4 is stable, it's the successor of go micro v3 and includes latest bug fixings, so it can be used in production. The main change of go micro v4 is it used new vanity url, it could a breaking changes when migrate from v2/v3. Go micro v3 is also stable, but only accept bug fixings.

- Thank you for your reply. 
- I'm preparing to write an `e-commerce platform(mall)` based on a microservices architecture. If `go-micro v4` is stable, then I can just develop the platform based on it. 
- It may be `open sourced` after some of the features have been developed. Primarily based on `go + flutter`. 
- Initially, I tried `go-zero, go-kratos`, but I encountered some problems. 
- Because I used to be familiar with `go-micro v2`. I personally prefer `the design of go-micro`, since go-micro is actively maintained, there is no reason not to use it. 
- Also, are there any official communication groups for go-micro now? discord? slack? telegram? wechat? Anything is fine.

**Reply by xpunch** [reply]
I've build a micro service IoT platform based on go micro, and it's very stable on our production environment. There is no official communication group right now, maybe we'll create a slack group later.

**Reply by hhstore** [reply]
- cool.
- It's good to use `slack`. Also consider `discord`. (Slack needs to pay for a lot of features, discord's free features are enough to use) 
- There are many official groups of projects that use `discord`, like `rust, flutter`.
- I can also be involved in some `go-micro` community development works. 
- I used to be familiar with the `go-micro v2` source codes. 
- About  `go-micro v4`, I need to catch up with the latest changes. 😄

**Reply by xpunch** [reply]
I've create a slack workspace, you can join it if you need. [Slack](https://join.slack.com/t/newworkspace-kpb7563/shared_invite/zt-175aaev1d-iHExPTlfxvfkOeeKLIYEYw) will be added to some community led projects too.

**Reply by Jackneill** [reply]
Regarding the community channel: I'd also vote against slack in favour something else like discord (m3o is also on discord). Zulip is also worth conidering (rust uses that).

**Reply by xpunch** [reply]
I've create a [discord server](https://discord.gg/qV3HvnEJfB) too.

**Reply by Davincible** [reply]
@hhstore 

> Initially, I tried go-zero, go-kratos, but I encountered some problems.

For my own research, I'm curious what problems you encountered with those platforms

**Comment by asim** [comment]
We rise from the ashes to come and take everyone's microservices platforms by storm.

This will largely be a community led effort. I've made @xpunch an admin of the new go-micro org and I'll let him decide who else gets added, likely others who were maintainers here. The main project will remain here, the interfaces are largely stable and there's not much changes to happen in them, if anything I want to make sure that stays the case. BUT, we want to promote a HUGE ecosystem. And I think that can only be done by creating an org that's managed by the community itself. 

Thinks built or to be built:

- Plugins
- Examples
- Dashboard
- CLI
- Gateway
- Services
- ....

**Reply by xpunch** [reply]
I've move go-micro-dashboard into go-micro org, but there is also a dashboard repo, I think we should overwrite the dashboard repo with go-micro-dashboard project, cause it also contains the frontend project inside it, then rename go-micro-dashboard to dashboard, and I'll update the go.mod and related codes.

**Reply by asim** [reply]
OK go ahead. Nothing should have go-micro- prefix, that's the org itself. Everything I just pulled directly from this repo.

**Comment by asim** [comment]
Just an update. Plugins and everything else have moved to the new org. We already have two maintainers and looking for more in future. We can see ecosystems emerging with go-kit, go-zero, go-kratos and go-micro should be included in that. 

Go Micro is now in v4 with the goal of continuing to evolve and stabilise interfaces for distributed systems development with one reference implementation colocated in the repo. The rest will be left to the community ecosystem.
